examples/end-to-end-testing/flake.nix

```diff
@@ -28,10 +28,12 @@
           pname = "example-e2e";
           version = "0.1";
           doCheck = false;
-          nativeBuildInputs = lib.optionals pkgs.stdenv.isDarwin [
+          nativeBuildInputs = lib.optionals pkgs.stdenv.isDarwin (with pkgs.darwin.apple_sdk.frameworks; [
             pkgs.libiconv
-            pkgs.darwin.apple_sdk.frameworks.Security
-          ];
+            CoreFoundation
```

Comment:
**question (non-blocking):** Do you know what are these for and what are a good defaults? AFAIR I always have to add `libiconv` and `Security`, but I don't even know why and what is the most commonly used set.

Reply:
I think darwin builds usually complain if you don't have them (I always try without them and add them in if the build fails :man_shrugging: )

Reply:
I figured it out from the [logs](https://github.com/ipetkov/crane/actions/runs/6490015759/job/17625187508#step:5:2581), no idea what exactly it does.

![image](https://github.com/ipetkov/crane/assets/31388299/437fce27-5e95-4bf5-9715-3c36b304a2d7)